source/game.ai.bmx

```diff
@@ -786,17 +786,17 @@ Type TLuaFunctions Extends TLuaFunctionsBase {_exposeToLua}
 		Return TProgrammeLicence[](obj)
 	End Method
 
-	'required until brl.reflection correctly handles "float parameters" 
-	'in debug builds (same as "doubles" for 32 bit builds)
-	'GREP-key: "brlreflectionbug"
-	Method CopyBasicAudienceAttractionString:TAudienceAttraction(attraction:TAudienceAttraction, multiplyFactor:String)
```

Comment:
Das Entfernen dieser API macht den Vergleich mit älteren KI-Implementierungen schwerer. Man muss dann nämlich immer auch den noch nicht angepassten Altcode an die neue API anpassen.

Reply:
Klar kann ich das jetzt wieder einfuehren ... aber dann landen wir schnell in der Spirale, ja nie "aufraeumen" zu koennen, ohne das Altdaten damit nicht mehr klarkommen.

Wo ziehen wir da also die Grenze? 

Wenn Dir diese Funktion besonders wichtig erscheint, koennen wir sie natuerlich wieder reinnehmen ... mit obigem im Hinterkopf.

Reply:
Ich wollte nur darauf hinweisen, dass jedes Ändern der von LUA verwendeten API Gefahren birgt (Regression oder eben beim Performancevergleich zwischen Versionen). Erschwerend kommt hinzu, dass es ja nur Laufzeitprobleme sind, d.h. der Code muss auch wirklich durchlaufen werden (und man muss darauf achten, ob und dass es knallt).

Ich frage mich auch, ob dasselbe Problem (float-Parameter im debug-Modus) vielleicht sogar noch bei direkten Aufrufen in Blitzmaxobjekte besteht, nur dass es nicht auffällt, weil ja irgendein Wert für die Berechnung verwendet wird.